packages/block-editor/src/store/selectors.js

```diff
@@ -1138,12 +1138,18 @@ export function isCaretWithinFormattedText( state ) {
 }
 
 /**
- * Returns the insertion point, the index at which the new inserted block would
- * be placed. Defaults to the last index.
+ * Returns the insertion point. This will be:
  *
- * @param {Object} state Editor state.
+ * 1) The insertion point manually set using setInsertionPoint() or
+ *    showInsertionPoint(); or
+ * 2) The point after the current block selection, if there is a selection; or
+ * 3) The point at the end of the block list.
+ *
+ * Components like <Inserter> will default to inserting blocks at this point.
+ *
+ * @param {Object} state Global application state.
  *
- * @return {Object} Insertion point object with `rootClientId`, `index`.
+ * @return {Object} Insertion point object with `rootClientId` and `index`.
```

Comment:
Previously in this PR `getBlockInsertionPoint` also returned `isVisible`, but I've removed that as it seems to be unused.